### ISSUE TITLE
Fast refresh

### ISSUE THREAD

**Issue by lifeparticle** [lifeparticle/binarytree#216]

**Comment by lifeparticle** [comment]
1. Organise the routing using parent and children concept

**Comment by ashik-75** [comment]
In the `items.contant.tsx` file, I encountered a hurdle while trying to render the Lucide Icon component. To achieve this, I needed to use the `.tsx` extension for the file to support TypeScript and JSX syntax. However, the challenge arose because I wanted to export a constant from this file. According to the rules of Hot Module Refresh (HMR) in React, exporting constants in a React component file is not allowed.

This situation resulted in a contradiction between our needs and expectations. Although I came across some open-source code where this rule seems to be avoided, the implementation remained similar to ours.

To summarize, the requirement to use `.tsx` extension to render Lucide Icon components conflicted with the need to export a constant in the React component file, as per HMR rules. Finding a workaround for this issue became crucial to solve the `react-refresh/only-export-components` warnings.
